### ISSUE TITLE
Multiple StackingEstimators - interpretation?

### ISSUE THREAD

**Issue by KirkDCO** [EpistasisLab/tpot#1101]
In running a TPOT job, the best pipeline has multiple StackingEstimators in sequence:

    StackingEstimator(estimator=LassoLarsCV(normalize=True)),
    StackingEstimator(estimator=AdaBoostRegressor(learning_rate=0.5, loss="square", n_estimators=100)),
    StackingEstimator(estimator=RidgeCV()),
    SGDRegressor(alpha=0.001, eta0=1.0, fit_intercept=True, l1_ratio=0.75, learning_rate="invscaling", loss="huber", penalty="elasticnet", power_t=0.1)

In this situation, how do I interpret what is actually happening?  I see two possibilities:

One possibility is that LassoLarsCV, AdaBoostRegressor, and RidgeCV make predictions where are appended to the data matrix and SGDRegressor trains a model with p+3 data columns.

More likely (I think), LassoLarsCV uses a matrix of p columns and its prediction is appended, AdaBoostRegressor uses a matrix of p+1 (data plus LassoLarsCV prediction), RidgeCV uses a matrix of p+2, and finally SGDRegressor uses a matrix of p+3.

If the second interpretation that a progressive stacking occurs, is it possible for TPOT to generate the first interpretation in which 3 different predictors contribute to a matrix with p+3 columns but each of those has only a p columns to start with?

**Comment by weixuanfu** [comment]
Yes, the second interpretation is what happened in that pipeline. But TPOT does not support the 1st interpretation so far.

**Comment by KirkDCO** [comment]
Thanks for the super fast response!  

I'm really enjoying using TPOT and finding some very interesting things.  Nice work!
